tern/classes/docker_image.py

```diff
@@ -71,14 +66,10 @@ def to_dict(self, template=None):
     def get_image_option(self):
         '''Check to see which value was used to init the image object
         Return the value that was used. If neither one was used raise
-        NameError. If both were used return the id'''
-        if self.repotag is not None and self.image_id is not None:
-            return self.image_id
+        NameError.'''
         if self.repotag is not None:
             return self.repotag
-        if self.image_id is not None:
-            return self.image_id
-        raise NameError("Image object initialized with no repotag or ID")
+        raise NameError("Image object initialized with no repotag")
```

Comment:
This whole function can be deleted. There is already a check for no repotag. You will then need to make a change in `load_image()`:
```
container.extract_image_metadata(option) --> container.extract_image_metadata(self.repotag)
```